price-estimator/README.md

```diff
@@ -0,0 +1,25 @@
+## API
+
+The service exposes the following endpoints:
+
+### estimated buy amount
+
+GET `/markets/<base_token_id>-<quote_token_id>/estimated-buy-amount/<sell_amount_in_quote_token>`
+
+* `base_token_id` and `quote_token_id` are the token ids as in the smart contract.
+* `sell_amount_in_quote_token` is positive integer.
+
+Example Request: `/markets/0-1/estimated-buy-amount/2`
+
+Example Response:
+
+```json
+{
+    "baseTokenId": "0",
+    "quoteTokenId": "1",
+    "buyAmountInBase": "3",
+    "sellAmountInQuote": "2",
+}
+```
```

Comment:
It's not necessary but this is how the nodejs price estimator works: https://github.com/gnosis/dex-price-estimator/issues/43 .

Reply:
The reason for returning it was to make it crystal clear what value is what as from the name of the name URL `estimatedBuyAmount/<someNumber>` it wasn't necessarily clear what are the semantics of the parameter and the returned number.

This duplication is based on the Kyber API: https://developer.kyber.network/docs/API_ABI-RESTfulAPI/#buy_rate

Reply:
Maybe we should use realistic numbers here (maybe trading 10**18 DAI vs 10**6 USDC as they have different decimals)

Reply:
I'd like to keep this example response simple because it means to only show the json structure but if the unit in which the buy amount is given is unclear then we can add another explanatory sentence.

Reply:
Usually API docs give real world examples of their usage even when the goal is to only document the structure (cf. web3 docs, kyber link above). I think people subconsciously interpret semantics when reading the docs and by seeing small numbers readers might think we are returning base units (ETH rather than WEI)

Reply:
> trading 10**18 DAI vs 10**6 USDC

:rofl: pwnd by Markdown syntax - this gets me every time. It took me a second that you meant 10\*\*18 and 10\*\*6

Reply:
Could you give the full json example you prefer?

Reply:
> Could you give the full example you prefer?

I would say, just run the service and make a query for buying 1e18 WETH with DAI - should be around 200e18. That way you get a real world example so you know what order of magnitude to expect for these numbers.